torch/testing/_internal/common_distributed.py

```diff
@@ -956,6 +956,19 @@ def perThreadTearDown(self):
     def world_size(self) -> int:
         raise RuntimeError("world size not implemented")
 
+    @property
+    def rank(self) -> int:
+        return c10d.get_rank()
+
+    def assertEqualRank0(self, x, y):
```

Comment:
can we make this like `def assertEqualOnRank(self, x, y, rank=0)`?

Maybe also add a `assertNotEqualOnRank` too?